Td 150 course enrollment

RitaKruglova opened the pull request:
**Изменения**
- Сверстала страницу курса
- Сверстала 3 попапа
- Написала логику календаря
- Сделала запросы на запись на мастер-класс

RitaKruglova committed:
feat: add marking for CalendarPopup

RitaKruglova committed:
fix: change ModalOverlay component

RitaKruglova committed:
feat: add useModalClose hook

RitaKruglova committed:
feat: add createPortal for calendar popup

RitaKruglova committed:
feat: add styles for CalendarPopup for tablet

RitaKruglova committed:
feat: add styles for CalendarPopup for mobile

RitaKruglova committed:
fix: correct overlay for CalendarPopup

RitaKruglova committed:
feat: add IconInfo component and RegistrationLessonPopup marking

RitaKruglova committed:
feat: add styles for RegistrationLessonPopup

RitaKruglova committed:
feat: add ApplicationAcceptedPopup marking

RitaKruglova committed:
merge dev

RitaKruglova committed:
feat: add styles for mobile for ApplicationAcceptedPopup component

RitaKruglova committed:
refactor: replace buttons in popups by Button component

RitaKruglova committed:
feat: add CalendarDsy component and styles for one

RitaKruglova committed:
feat: add functionality of calendar

ReneNorth committed:
add teacher info to masterclass types endpoint

ReneNorth committed:
change teacher_id in serializer from char field to ID field

RitaKruglova committed:
feat: use info of selected lesson in CalendarPopup

ReneNorth committed:
make filed comment in feedback required

RitaKruglova committed:
fix: change course to masterclass in course page

RitaKruglova committed:
Merge branch 'TD-150-Course-enrollment-' of https://github.com/ReneNorth/Mosaic-courses-website into TD-150-Course-enrollment-

RitaKruglova committed:
fix: create function foller in utils and move calculateLessonDuration there

RitaKruglova committed:
fix: change course to masterclass in popups and fix logic

RitaKruglova committed:
feat: add state and handler for checkbox in useFormValidation

RitaKruglova committed:
Revert "feat: add state and handler for checkbox in useFormValidation"

This reverts commit 063d122191cb76f2ad5f32264d57103506715e4f.

RitaKruglova committed:
fix: functionality of header links

ReneNorth committed:
adjusted feedback test

ReneNorth commented:
@RitaKruglova 
Рита, едет вёрстка при клике на видео (на Mac OS, десктоп, браузер Chrome)
подробнее в задаче, там приложено видео https://tessera-mosaic.atlassian.net/jira/software/projects/TD/boards/1?selectedIssue=TD-223

zigfrei commented on frontend/src/components/ApplicationAcceptedPopup/ApplicationAcceptedPopup.module.scss:
```diff
@@ -0,0 +1,104 @@
+  border-radius: 20px;
+  background-color: var(--color-white);
+  margin-top: 80px;
+  font-family: "Raleway", "Arial", sans-serif;
```
Шрифт лучше через кастомную переменную

zigfrei commented on frontend/src/components/Calendar/Calendar.jsx:
```diff
@@ -0,0 +1,175 @@
+    setActiveDays(activeDates);
+  }, [currentCourse]);
+
+  const generateDays = (date) => {
+    const year = date.getFullYear();
+    const month = date.getMonth();
+    const firstDay = new Date(year, month, 1);
+    const lastDay = new Date(year, month + 1, 0);
+
+    const daysOfWeek = [7, 1, 2, 3, 4, 5, 6];
+    const firstWeekDay = daysOfWeek[firstDay.getDay()];
+
+    const previousDays = Array.from({ length: firstWeekDay - 1 }, (_, i) => new Date(year, month, -i)
+      .getDate()).reverse();
+    const currentDays = Array.from({ length: lastDay.getDate() }, (_, i) => i + 1);
+    const nextDays = Array.from({ length: 42 - previousDays.length - currentDays.length }, (_, i) => i + 1);
+
+    return { previous: previousDays, current: currentDays, next: nextDays };
+  };
```
Наверное это лучше вынести в хелпер? и на него надо написать тест

zigfrei commented on frontend/src/components/Calendar/Calendar.jsx:
```diff
@@ -0,0 +1,175 @@
+    dispatch(setSelectedLesson(lesson));
+  };
+
+  const monthAndYear = `${['Январь', 'Февраль', 'Март', 'Апрель', 'Май', 'Июнь', 'Июль',
+    'Август', 'Сентябрь', 'Октябрь', 'Ноябрь', 'Декабрь'][date.getMonth()]} ${date.getFullYear()}`;
```
Перечень месяцев на русском наверное надо в константу, да и сам метод похож на хэлпер

zigfrei commented on frontend/src/components/Calendar/Calendar.module.scss:
Ты `@media` запросы пишешь внутри классов. Для меня например удобно если там при определленом расширении надо что-то подправить для конкретного класса то можно в нем прямо и написать. Но если у тебя много изменений для например 1320px может логичнее это всетаки вынести в отдельный блок? и там применить все изменения для этого расширения?

zigfrei commented on frontend/src/components/CalendarPopup/CalendarPopup.module.scss:
```diff
@@ -0,0 +1,140 @@
+.title {
+  margin-top: 48px;
+  font-family: 'Raleway', 'Arial', sans-serif;
```
Шрифт в переменную

zigfrei commented on frontend/src/components/CalendarPopup/CalendarPopup.module.scss:
Тоже по `@media`

zigfrei commented on frontend/src/components/OtherCourse/OtherCourse.module.scss:
```diff
@@ -134,16 +85,81 @@
   margin-top: 60px;
+}
+
+@media screen and (max-width: #{global.$breakpoint-tablet-for-course-page}) {
```
Как-то выглядит это по хуже чем было `#{global.$breakpoint-tablet-for-course-page}` по этому не совсем понял зачем ты это сделала

zigfrei commented on frontend/src/components/OtherCourse/OtherCourse.module.scss:
```diff
@@ -134,16 +85,81 @@
   }
+}
+
+@media screen and (max-width: #{global.$breakpoint-tablet}) {
```
Аналогично `#{global.$breakpoint-tablet}` смотрится жутко

zigfrei commented on frontend/src/components/RegistrationLessonPopup/RegistrationLessonPopup.jsx:
```diff
@@ -0,0 +1,116 @@
+
+  const getFormattedDate = (dateString) => {
+    const date = new Date(dateString);
+    const months = [
+      'января', 'февраля', 'марта', 'апреля', 'мая', 'июня',
+      'июля', 'августа', 'сентября', 'октября', 'ноября', 'декабря',
+    ];
```
Надо вынести в константу

zigfrei reviewed (commented):
#### Общее замечание:
- Может вынестии 1320 в константу?
- надо в сторибук переместить календарь
П.с. Локально код не проверял и проект не поднимал с этими правками

RitaKruglova committed:
fix: add checking for IconInfo

RitaKruglova committed:
Merge remote-tracking branch 'origin/TD-150-Course-enrollment-' into TD-150-Course-enrollment-

RitaKruglova committed:
fix: review's comments

RitaKruglova committed:
feat: add test for generateDays.js

RitaKruglova committed:
fix:
calculate the scroll width and subtract it from the screen width when the popup is open

ReneNorth committed:
Merge pull request #119 from ReneNorth/hotfix-update-pip-install-dependency

Hotfix update pip install dependency

ReneNorth commented:
@zigfrei ревью/комментов судя по всему больше не будет, поэтому я мерджу как есть
если будут какие комменты всё же - сделаем в рамках отдельной задачи

RitaKruglova committed:
Merge branch 'dev' into TD-150-Course-enrollment-
